src/storage/hummock_sdk/src/filter_key_extractor.rs

```diff
@@ -298,7 +302,14 @@ impl FilterKeyExtractorManagerInner {
             }
 
             if !table_id_set.is_empty() {
-                notified.await;
+                timeout(ACQUIRE_TIMEOUT, notified)
```

Comment:
How about only enabling this on `#[cfg(debug_assertions)]` as we panic here. I'm not sure what will happen in production. 😢

Reply:
> How about only enabling this on `#[cfg(debug_assertions)]` as we panic here? I'm not sure what will happen in production. 😢 cc @yezizp2012

If stuck bug happen in production without panic, its difficult for us to check it ? 

😢 how about introduce the `async-stack-trace` in production and panic on `debug_assertions ` ?

Reply:
Yes, we've planned to enable the async stack tracing in production.

Reply:
If stuck, a recovery progress should have been triggered already if it's not caused by any bugs. I think panic here is acceptable (only on `#[cfg(debug_assertions)]`), and I think eventually we should have a way to notice this happened in production. An alarm system based on logs may helps.